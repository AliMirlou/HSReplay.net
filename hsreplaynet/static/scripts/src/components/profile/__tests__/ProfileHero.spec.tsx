import React from "react";
import ProfileHero from "../ProfileHero";
import renderer from "react-test-renderer";

describe("ProfileHero", () => {
	test("does render with without name", () => {
		const component = renderer.create(<ProfileHero cardId={"HERO_07"} />);
		const tree = component.toJSON();
		expect(tree).toBeDefined();
		expect(tree.children.length).toBe(1);
		expect(tree.children[0].children.length).toBe(2);
		expect(tree.children[0].children[0].props.src).toBe(
			"https://art.hearthstonejson.com/v1/256x/HERO_07.jpg",
		);
		expect(tree.children[0].children[1].props.src).toBe(
			"/static/images/profile-hero-frame.png",
		);
		expect(tree).toMatchSnapshot();
	}),
		test("does render with name", () => {
			const component = renderer.create(
				<ProfileHero cardId={"HERO_07"} name="Test Name" />,
			);
			const tree = component.toJSON();
			expect(tree).toBeDefined();
			expect(tree.children.length).toBe(2);
			expect(tree.children[1].children[0]).toBe("Test Name");
			expect(tree).toMatchSnapshot();
		});
});
