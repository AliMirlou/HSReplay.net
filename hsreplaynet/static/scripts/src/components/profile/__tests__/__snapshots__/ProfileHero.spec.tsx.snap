// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHero does render with name 1`] = `
<figure
  className="profile-hero"
>
  <div>
    <img
      className="profile-hero-art"
      src="https://art.hearthstonejson.com/v1/256x/HERO_07.jpg"
    />
    <img
      className="profile-hero-frame"
      src="/static/images/profile-hero-frame.png"
    />
  </div>
  <figcaption>
    Test Name
  </figcaption>
</figure>
`;

exports[`ProfileHero does render with without name 1`] = `
<figure
  className="profile-hero"
>
  <div>
    <img
      className="profile-hero-art"
      src="https://art.hearthstonejson.com/v1/256x/HERO_07.jpg"
    />
    <img
      className="profile-hero-frame"
      src="/static/images/profile-hero-frame.png"
    />
  </div>
</figure>
`;
